Mathlib/Data/Matroid/Minor/Basic.lean

```diff
@@ -0,0 +1,289 @@
+
+lemma IsCircuit.of_delete {C : Set α} (h : (M ＼ D).IsCircuit C) : M.IsCircuit C :=
+  (delete_isCircuit_iff.1 h).1
+
+lemma circuit_iff_delete_of_disjoint {C : Set α} (hCD : Disjoint C D) :
+    M.IsCircuit C ↔ (M ＼ D).IsCircuit C :=
+  ⟨fun h ↦ delete_isCircuit_iff.2 ⟨h, hCD⟩, fun h ↦ h.of_delete⟩
+
+@[simp]
+lemma delete_closure_eq (M : Matroid α) (D X : Set α) :
+    (M ＼ D).closure X = M.closure (X \ D) \ D := by
+  rw [← restrict_compl, restrict_closure_eq', sdiff_sdiff_self, bot_eq_empty, union_empty,
+    diff_eq, inter_comm M.E, ← inter_assoc X, ← diff_eq, closure_inter_ground,
+    ← inter_assoc, ← diff_eq, inter_eq_left]
+  exact diff_subset.trans (M.closure_subset_ground _)
+
+@[simp]
+lemma delete_loops_eq (M : Matroid α) (D : Set α) : (M ＼ D).loops = M.loops \ D := by
+  simp [loops]
+
+@[simp]
+lemma delete_empty (M : Matroid α) : M ＼ ∅ = M := by
+  rw [delete_eq_self_iff]
+  exact empty_disjoint _
+
+lemma delete_delete_diff (M : Matroid α) (D₁ D₂ : Set α) : M ＼ D₁ ＼ D₂ = M ＼ D₁ ＼ (D₂ \ D₁) :=
+  by simp
+
+instance delete_finitary (M : Matroid α) [Finitary M] (D : Set α) : Finitary (M ＼ D) := by
+  change Finitary (M ↾ (M.E \ D)); infer_instance
+
+lemma Coindep.delete_isBase_iff (hD : M.Coindep D) :
+    (M ＼ D).IsBase B ↔ M.IsBase B ∧ Disjoint B D := by
+  rw [Matroid.delete_isBase_iff]
+  refine ⟨fun h ↦ ?_, fun h ↦ ?_⟩
+  · have hss := h.subset
+    rw [subset_diff] at hss
+    have hcl := h.isBasis_closure_right
+    rw [hD.closure_compl, isBasis_ground_iff] at hcl
+    exact ⟨hcl, hss.2⟩
+  exact h.1.isBasis_ground.isBasis_subset (by simp [subset_diff, h.1.subset_ground, h.2])
+    diff_subset
+
+lemma Coindep.delete_rankPos [M.RankPos] (hD : M.Coindep D) : (M ＼ D).RankPos := by
+  rw [rankPos_iff, hD.delete_isBase_iff]
+  simp [M.empty_not_isBase]
+
+lemma Coindep.delete_spanning_iff (hD : M.Coindep D) :
+    (M ＼ D).Spanning S ↔ M.Spanning S ∧ Disjoint S D := by
+  simp only [spanning_iff_exists_isBase_subset', hD.delete_isBase_iff, and_assoc, delete_ground,
+    subset_diff, and_congr_left_iff, and_imp]
+  refine fun hSE hSD ↦ ⟨fun ⟨B, hB, hBD, hBS⟩ ↦ ⟨B, hB, hBS⟩, fun ⟨B, hB, hBS⟩ ↦ ⟨B, hB, ?_, hBS⟩⟩
+  exact hSD.mono_left hBS
+
+end Delete
+
+section Contract
+
+/-- The contraction `M ／ C` is the matroid on `M.E \ C` in which a set `I ⊆ M.E \ C` is independent
+if and only if `I ∪ J` is independent, where `J` is an arbitrarily chosen basis for `C`.
+It is also equal to `(M✶ ＼ C)✶`, and is defined this way so we don't have to give
+a separate proof that it is actually a matroid.
+(Currently the proof it has the correct independent sets is TODO. ) -/
+def contract (M : Matroid α) (C : Set α) : Matroid α := (M✶ ＼ C)✶
+
+/-- `M ／ C` refers to the contraction of a set `C` from the matroid `M`. -/
+infixl:75 " ／ " => Matroid.contract
+
+lemma dual_delete_dual (M : Matroid α) (X : Set α) : (M✶ ＼ X)✶ = M ／ X := rfl
+
+@[simp]
+lemma delete_dual (M : Matroid α) (X : Set α) : (M ＼ X)✶ = M✶ ／ X := by
```

Comment:
I think the naming convention here would say `dual_delete`, similarly for others.